tests/basis_test.py

```diff
@@ -3,328 +3,274 @@
 import numpy
 from scipy import special
 
-from smolyay.basis import (BasisFunction, ChebyshevFirstKind,
-        ChebyshevSecondKind, BasisFunctionSet, NestedBasisFunctionSet)
-
-@pytest.fixture
-def expected_points_3_set():
-    """extrema for exactness = 8"""
-    return [0, -1.0, 1.0, -1/numpy.sqrt(2), 1/numpy.sqrt(2),
-            -numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75),
-            -numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75),
-            numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75),
-            numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75)]
-
-def test_cheb_initial_zero():
-    """test degree of zero"""
-    f = ChebyshevFirstKind(0)
-    assert f.n == 0
-    assert f.points == [0]
-
-def test_cheb_initial_1():
-    """test initial when degree is 1"""
-    f = ChebyshevFirstKind(1)
-    assert f.n == 1
-    assert f.points == [-1, 1]
-
-def test_cheb_initial_2():
-    """test initial when degree is 2"""
-    f = ChebyshevFirstKind(2)
-    assert f.n == 2
-    assert numpy.allclose(f.points,[-1, 0, 1],atol=1e-10)
-
-def test_cheb_initial_4():
-    """test initial when degree is 4"""
-    expected_points = [-1, -1/numpy.sqrt(2), 0, 1/numpy.sqrt(2), 1]
-    f = ChebyshevFirstKind(4)
-    assert f.n == 4
-    assert numpy.allclose(f.points,expected_points,atol=1e-10)
-
-def test_cheb_initial_8():
-    """test initial when degree is 8"""
-    expected_points = [-1, -numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75),
-            -1/numpy.sqrt(2), -numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75), 0,
-            numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75), 1/numpy.sqrt(2),
-            numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75), 1]
-    f = ChebyshevFirstKind(8)
-    assert f.n == 8
-    assert numpy.allclose(f.points,expected_points,atol=1e-10)
-
-def test_cheb_call_degree_0_1():
-    """Chebyshev polynomial degree 0 is always 1 and degree 1 returns input"""
-    f0 = ChebyshevFirstKind(0)
-    f1 = ChebyshevFirstKind(1)
-    for i in [-1, -0.5, 0, 0.5, 1]:
-        assert f0(i) == 1
-        assert f1(i) == i
-
-def test_cheb_call_random_points():
+import smolyay
+
+
+def test_cheb_initial():
+    """test degrees returns correctly"""
+    f2 = smolyay.basis.ChebyshevFirstKind(2)
+    assert f2.degree == 2
+    assert isinstance(f2.degree, int)
+    assert numpy.array_equal(f2.domain, [-1, 1])
+    f2.degree = 3
```

Comment:
Maybe you should assign this as a float to make sure it is silently converted to an `int`?
```suggestion
    f2.degree = float(3)
```